File: code-generator/src/main/java/io/github/potjerodekool/openapi/ParameterLocation.java
package io.github.potjerodekool.openapi;

import java.util.Arrays;

/**
 Enumeration of where a parameter can be located.
 */
public enum ParameterLocation {
    PATH("path"),
    QUERY("query"),
    HEADER("header"),
    COOKIE("cookie");

    private final String value;

    ParameterLocation(final String value) {
        this.value = value;
    }

    public static ParameterLocation parseIn(final String in) {
        return Arrays.stream(values())
                .filter(pl -> pl.value.equals(in))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException(in));
    }
}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/ParameterLocation.kt
package io.github.potjerodekool.openapi.common

import java.util.*

/**
 * Enumeration of where a parameter can be located.
 */
enum class ParameterLocation(private val value: String) {
    PATH("path"),
    QUERY("query"),
    HEADER("header"),
    COOKIE("cookie");

    fun value(): String {
        return value
    }

    companion object {
        fun parseIn(`in`: String): ParameterLocation {
            return Arrays.stream(values())
                .filter { pl: ParameterLocation -> pl.value == `in` }
                .findFirst()
                .orElseThrow { IllegalArgumentException(`in`) }
        }
    }
}
